aip/client-libraries/4234.md

```diff
@@ -0,0 +1,92 @@
+---
+id: 4234
+state: draft
+created: 2021-02-10
+---
+
+# Common service client mix-ins
+
+An API platform often has utility services that are shared amongst the discrete
+product services. These common services each have a centrally-defined surface,
+but individual instances are hosted alongside each product service. Thus, the
+surfaces of each common services hosted by the product service are "mixed in" at
+runtime.
+
+
+## Guidance
+
+Client libraries **may** provide client methods for the common services declared
+to be mixed-in by a product service. Such methods improve the user experience by
+presenting the mix-in methods from a client configured to communicate with the
+product service that hosts the mix-in service.
+
+Should client libraries support mix-in services, the common services that
+**must** be supported include:
+
+- `google.cloud.location.Locations`
+- `google.iam.v1.IAMPolicy`
+- `google.longrunning.Operations`
+
+**Note:** The list of supported mix-in services and the mix-in services
+themselves change infrequently. Any new common service to be supported should be
+added to the list here and generators should be updated.
+
+A mix-in service **must** be declared in the [`apis`] in a
+[`google.api.Service`] to be generated into a client library. Furthermore, it
```

Comment:
For clarity, maybe flip the sentence: "To be generated into a client library, a mix-in service **must** ..."

Reply:
I like it! Done.